### PR TITLE
Modify example to add bound constraints

### PR DESCRIPTION
**Pull request by mzaffalon**
Following discussion in issue #93.

### CODE REVIEW

**Comment by pkofod** [comment]
Thanks for the PR. I was actually thinking of having a separate example below the one without bounds. I'd be a bit afraid that people would think that bounds were required given the current example. Does that make sense?

**Comment by mzaffalon** [comment]
You mean a completely different one or something like
```
# original example 
fit = curve_fit(model, xdata, ydata, p0)

# upper and/or lower bounds on the free parameters can be specified
# bounded and unbouded variables can be mixed by setting `-Inf` if no lower bounds
# must be enforced for that variable and similarly for `+Inf`
fit_bounds = curve_fit(model, xdata, ydata, p0, lower=lb, upper=up)
```

**Comment by pkofod** [comment]
Yes :+1:

File: README.md [hunk 1]
```diff
@@ -31,19 +31,22 @@ using LsqFit
...
 #	fit.jacobian: estimated Jacobian at solution
-@assert all(lb .<= fit.param .<= ub)
+
+lb = [1.1, -0.5]
+ub = [1.9, Inf]
+# Optional upper and/or lower bounds on the free parameters can be passed as an argument.
+# Bounded and unbouded variables can be mixed by setting `-Inf` if no lower bounds
+# is to be enforced for that variable and similarly for `+Inf`
+fit_bounds = curve_fit(model, xdata, ydata, p0, lower=lb, upper=ub)
+@assert all(lb .<= fit_bounds.param .<= ub)
```
Comment (mzaffalon) [inline README.md]: I am not sure we gain much from this assert. Should I remove it?

Comment (pkofod) [reply README.md]: Yes, just remove it.

**Comment by pkofod** [comment]
Are you happy with it now?

**Comment by mzaffalon** [comment]
That's kind of you to ask! I am. Are you?

**Comment by pkofod** [comment]
I am ! Thanks.

**Pull request merged**